### DISCUSSION TITLE
using add_csv function in leafmap.kepler

### DISCUSSION THREAD

**Discussion by curtis18** [opengeos/leafmap#543] [Q&A] [answered]
I tried to to use leafmap.kepler, it is successful to create an empty map. However, when I run the demo code, only empty spacing is loaded. May I know how to debug or check the log in order to know what the problem is? If I remove add_csv function, it will be fine, but it is not what I want.

m = leafmap.Map(center=[37.7621, -122.4143], zoom=12)
in_csv = (
    'https://raw.githubusercontent.com/opengeos/leafmap/master/examples/data/hex_data.csv'
)
config = 'https://raw.githubusercontent.com/opengeos/leafmap/master/examples/data/hex_config.json'
m.add_csv(in_csv, layer_name="hex_data", config=config)

![image](https://github.com/opengeos/leafmap/assets/6743839/380fef73-f5c9-4e78-b219-191efe1c5931)

![image](https://github.com/opengeos/leafmap/assets/6743839/d2958298-b3c3-4e9a-b238-e99389a5cb9e)

leafmap  version 0.24.1 to 0.24.4

jupyter version
Selected Jupyter core packages...
IPython          : 8.15.0
ipykernel        : 6.25.2
ipywidgets       : 7.8.0
jupyter_client   : 7.4.9
jupyter_core     : 5.3.1
jupyter_server   : 2.7.3
jupyterlab       : 3.6.5
nbclient         : 0.8.0
nbconvert        : 7.8.0
nbformat         : 5.9.2
notebook         : 6.5.5
qtconsole        : not installed
traitlets        : 5.9.0

node version
v18.17.1

python version
Python 3.10.12

**Comment by giswqs** [comment, marked as answer]
The kepler mapping backend is no longer being developed. I would encourage you to use thei ipyleaflet backend. Is there any specific reason feature you want to use with Kelper that are not available through other backends?

**Reply by curtis18** [reply]
@giswqs Hi Dr. Wu, thanks for your reply. Do you mean the binding or wrapper of python is no longer being developed? I think the visual appealing of Kepler is good and it make use of WebGL. I want to use Kelper to implement 2.5D map as follows. I found Maplibre, similar to Leaflet, also supports 2.5D, but it does not seem to integrate with Leafmap. Apart from Kepler, do you have any recommendation?

![image](https://github.com/opengeos/leafmap/assets/6743839/91cace5f-dfe2-4281-89de-13bc63141ed2)


![image](https://github.com/opengeos/leafmap/assets/6743839/9e9feca2-9f9a-4e68-8df8-e3f8336c67dc)

**Reply by giswqs** [reply]
I don't used KeplerGL for a while. The KelperGL integration with leafmap has not been updated for a long time. You are welcome to look at the source code and adpat to your needs.

**Reply by curtis18** [reply]
@giswqs Thanks for your update. I will try to look at the source code.
